### PR TITLE
cmake: build PeakFlops only on x86

### LINKED ISSUE #64: unknown register name '%rdx' in 'asm'
With gcc-4.8.5, I get:
```
[ 30%] Built target Slice
/root/hpc-gitlab-runner/ecp-copa/cabana/builds/users/junghans/47b77dfd/0/ecp-copa/cabana/core/example/benchmark/Cabana_peakflops.cpp: In function 'void run()':
/root/hpc-gitlab-runner/ecp-copa/cabana/builds/users/junghans/47b77dfd/0/ecp-copa/cabana/core/example/benchmark/Cabana_peakflops.cpp:23:103: error: unknown register name '%rcx' in 'asm'
   asm volatile ("rdtscp;shlq $32,%%rdx;orq %%rdx,%%rax;movq %%rax,%0":"=q"(u)::"%rax", "%rdx", "%rcx");
                                                                                                       ^
/root/hpc-gitlab-runner/ecp-copa/cabana/builds/users/junghans/47b77dfd/0/ecp-copa/cabana/core/example/benchmark/Cabana_peakflops.cpp:23:103: error: unknown register name '%rdx' in 'asm'
/root/hpc-gitlab-runner/ecp-copa/cabana/builds/users/junghans/47b77dfd/0/ecp-copa/cabana/core/example/benchmark/Cabana_peakflops.cpp:23:103: error: unknown register name '%rax' in 'asm'
/root/hpc-gitlab-runner/ecp-copa/cabana/builds/users/junghans/47b77dfd/0/ecp-copa/cabana/core/example/benchmark/Cabana_peakflops.cpp:23:103: error: unknown register name '%rcx' in 'asm'
```

I guess we should add a check against old compilers!

### PR DESCRIPTION
**Pull request by junghans**
Fix #64

### CODE REVIEW

File: core/example/benchmark/CMakeLists.txt [hunk 1]
```diff
@@ -3,5 +3,7 @@ if(Cabana_ENABLE_Serial)
   target_link_libraries(NeighborListMDPerfTest cabanacore)
 endif()
 
-add_executable(PeakFlops Cabana_peakflops.cpp)
-target_link_libraries(PeakFlops cabanacore)
+if(CMAKE_SYSTEM_PROCESSOR MATCHES "(x86)|(X86)|(amd64)|(AMD64)")
```
Comment (dalg24) [inline core/example/benchmark/CMakeLists.txt]: Did you mean to have x86 OR amd?  Your PR subject says "only on x86"

Comment (dalg24) [reply core/example/benchmark/CMakeLists.txt]: Why do you have the patterns between parentheses?
Why do you provide uppercase variants?

Comment (dalg24) [inline core/example/benchmark/CMakeLists.txt]: Why are patterns between parentheses?
Why do you provide uppercase variants?

Comment (sslattery) [reply core/example/benchmark/CMakeLists.txt]: Good question - I am unfamiliar with the syntax here.

Comment (junghans) [reply core/example/benchmark/CMakeLists.txt]: Just copy-paste from stackoverflow!

Comment (dalg24) [reply core/example/benchmark/CMakeLists.txt]: You should have added a link to it for reference

**Comment by junghans** [comment]
ping @guangyechen @sslattery

**Comment by junghans** [comment]
@guangyechen @dalg24 @rfbird please review.

**Pull request merged**